pkg/apis/sessions/interfaces.go

```diff
@@ -10,3 +13,14 @@ type SessionStore interface {
 	Load(req *http.Request) (*SessionState, error)
 	Clear(rw http.ResponseWriter, req *http.Request) error
 }
+
+var ErrLockNotObtained = errors.New("lock: not obtained")
+var ErrNotLocked = errors.New("tried to release not existing lock")
+
+// Lock is an interface for controlling session locks
+type Lock interface {
+	Obtain(ctx context.Context, expiration time.Duration) error
+	Peek(ctx context.Context) (bool, error)
+	Refresh(ctx context.Context, expiration time.Duration) error
+	Release(ctx context.Context) error
+}
```

Comment:
Would be good to have comments on the functions in the interface so we know what they do/expected behaviour etc.

This will aid future lock implementors to understand the interface